docs/jax-101/05.1-pytrees.md

```diff
@@ -240,9 +259,14 @@ To solve this, we need to register our container with JAX by telling it how to f
 
 from typing import Tuple, Iterable
 
-def flatten_MyContainer(container) -> Tuple[Iterable[int], str]:
+def flatten_with_keys_MyContainer(container) -> Tuple[Iterable[int], str]:
   """Returns an iterable over container contents, and aux data."""
-  flat_contents = [container.a, container.b, container.c]
+  
+  # GetAttrKey is a common way to express an attribute key. Users are free
+  # to pick any other expression that fits their use cases the best.
+  flat_contents = [(jax.tree_util.GetAttrKey('a'), container.a),
+                   (jax.tree_util.GetAttrKey('b'), container.b),
+                   (jax.tree_util.GetAttrKey('c'), container.c)]
```

Comment:
WDYT about keeping the non-key API example (i.e. the existing example using `register_pytree-node`), and then appending the example which shows how to use `register_pytree_with_keys`? We don't want to un-document a lot of existing code. Moreover, this extended API is optional.

Reply:
Done.